WordPress/openverse
# Sync all pre-commit configs from this repo

## Fixes


Fixes #316 by @dhruvkb (this time for real)

## Description


This PR changes the pre-commit configuration into a Jinja template, and uses the templating feature newly added to the sync action to compile them into the right pre-commit files for the API, the catalog and the frontend.

## Testing instructions

To see the compiled configs, follow these steps:

0.  Install the required packages: `just install`
1.  Invoke the render recipe for pre-commit config:
	```shell
	# for the WordPress/openverse repo
	$ just render-precommit

	# for the WordPress/openverse-api and WordPress/openverse-catalog repos
	$ just render \
		.pre-commit-config.yaml.jinja \
		.pre-commit-config.backend.yaml \
		'{\"contains_js_code\":false}'

	# for the WordPress/openverse-frontend repo
	$ just render \
		.pre-commit-config.frontend.yaml.jinja \
		.pre-commit-config.frontend.yaml \
		'{\"contains_python_code\":false}'
	```
2.  Invoke the render recipe for Prettier config:
	```shell
	# for the WordPress/openverse, WordPress/openverse-api and WordPress/openverse-catalog repos
	just render-prettier

	# for the WordPress/openverse-frontend repo
	$ just render \
		prettier.config.frontend.js.jinja \
		prettier.config.frontend.js
	```

The `render-jinja.js` script simulates the template rendering code present in `BetaHuhn/repo-file-sync-action`.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
